### DISCUSSION TITLE
Maximum pickable objects

### DISCUSSION THREAD

**Discussion by AlxZb** [visgl/deck.gl#6146] [General]
Recently my data increased and I have many layer for this, the 80% has pickable = true but the console show me the following message about this interaction, I use a combination of icon, scatterplot and polygon layers:

`deck: Too many pickable layers, only picking the first 255`

And the question is... is 255 the maximum of pickable objects or I can change this configuration in some where?

Thanks for replies.

**Comment by kylebarron** [comment]
Deck.gl performs picking via "color-based" picking, where every object for every layer is assigned a color. 

From https://deck.gl/docs/developer-guide/custom-layers/picking#the-color-picking-technique:

> Each object in each layer gets its own picking color assigned. The picking color is determined using layer.encodePickingColor() that converts the index of a object of a given layer into a 3-byte color array (the color buffer allows us to distinguish between 16M unique colors per layer, and between 256 different layers)

It might be possible to override this to give just two bytes to the object index and two bytes to the layer index, but I think the standard advice is to consolidate the number of layers you use.
